reconciler/testing/table.go

```diff
@@ -285,44 +286,31 @@ func (r *TableRow) Test(t *testing.T, factory Factory) {
 		t.Errorf("Unexpected subresource updates occurred %#v", unexpected)
 	}
 
-	for i, want := range r.WantDeletes {
-		if i >= len(actions.Deletes) {
-			t.Errorf("Missing delete: %#v", want)
-			continue
-		}
-		got := actions.Deletes[i]
-		if got.GetName() != want.GetName() {
-			t.Errorf("Unexpected delete[%d]: %#v", i, got)
-		}
-		if !r.SkipNamespaceValidation && got.GetNamespace() != expectedNamespace {
-			t.Errorf("Unexpected delete[%d]: %#v", i, got)
+	// Build a set of unique strings that represent type-name{-namespace}.
+	// Adding type will help catch the bugs where several similarly named
+	// resources are deleted (and some should or should not).
+	gotDeletes := make(sets.String, len(actions.Deletes))
+	for _, w := range actions.Deletes {
+		n := w.GetResource().Resource + "~~" + w.GetName()
+		if !r.SkipNamespaceValidation {
+			n += "~~" + w.GetNamespace()
 		}
+		gotDeletes.Insert(n)
 	}
-	if got, want := len(actions.Deletes), len(r.WantDeletes); got > want {
-		for _, extra := range actions.Deletes[want:] {
-			t.Errorf("Extra delete: %s/%s", extra.GetNamespace(), extra.GetName())
+	wantDeletes := make(sets.String, len(actions.Deletes))
+	for _, w := range r.WantDeletes {
+		n := w.GetResource().Resource + "~~" + w.GetName()
+		if !r.SkipNamespaceValidation {
+			n += "~~" + w.GetNamespace()
 		}
+		wantDeletes.Insert(n)
 	}
-
-	for i, want := range r.WantDeleteCollections {
-		if i >= len(actions.DeleteCollections) {
-			t.Errorf("Missing delete-collection: %#v", want)
-			continue
-		}
-		got := actions.DeleteCollections[i]
-		if got, want := got.GetListRestrictions().Labels, want.GetListRestrictions().Labels; (got != nil) != (want != nil) || got.String() != want.String() {
-			t.Errorf("Unexpected delete-collection[%d].Labels = %v, wanted %v", i, got, want)
-		}
-		if got, want := got.GetListRestrictions().Fields, want.GetListRestrictions().Fields; (got != nil) != (want != nil) || got.String() != want.String() {
-			t.Errorf("Unexpected delete-collection[%d].Fields = %v, wanted %v", i, got, want)
+	if !gotDeletes.Equal(wantDeletes) {
+		if extra := gotDeletes.Difference(wantDeletes); len(extra) > 0 {
+			t.Errorf("Extra or unexpected deletes: %v", extra)
 		}
-		if !r.SkipNamespaceValidation && got.GetNamespace() != expectedNamespace {
-			t.Errorf("Unexpected delete-collection[%d]: %#v, wanted %s", i, got, expectedNamespace)
-		}
-	}
-	if got, want := len(actions.DeleteCollections), len(r.WantDeleteCollections); got > want {
-		for _, extra := range actions.DeleteCollections[want:] {
-			t.Errorf("Extra delete-collection: %#v", extra)
+		if missing := wantDeletes.Difference(gotDeletes); len(missing) > 0 {
+			t.Errorf("Missing deletes: %v", missing)
```

Comment:
These are going to have your weird encoding, right?  🤔

Reply:
Something like this
`trigger_test.go:653: Extra or unexpected deletes: map[subscriptions~~test-broker-test-trigger-test-trigger-uid~~test-namespace:{}]`
but I can change that to `missing.UnorderedList()` to streamline.

Reply:
Aaaannd... done.